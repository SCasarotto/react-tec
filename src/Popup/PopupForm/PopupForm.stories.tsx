import React from 'react';

import { Story, Meta } from '@storybook/react';

import { PopupForm, PopupFormProps } from './PopupForm';

export default {
  title: 'Popup/PopupForm',
  component: PopupForm,
  argTypes: {},
} as Meta;

const Template: Story<PopupFormProps> = (args) => <PopupForm {...args} />;

export const Default = Template.bind({});
Default.args = {
  visible: true,
  onClose: () => console.log('onClose'),
  onSubmit: () => console.log('onSubmit'),
  children: <p style={{ textAlign: 'center' }}>Some form goes here!</p>,
};
