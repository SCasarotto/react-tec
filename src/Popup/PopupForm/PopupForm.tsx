import React from 'react';

import { Form, FormProps } from '../../Form/Form';
import { PopupProps } from '../Popup';
import {
  Popup,
  FormContent,
  ButtonContainer,
  LeftButton,
  RightButton,
} from './styledComponents';

export interface PopupFormProps extends PopupProps, FormProps {
  onClose(): void;
  onCloseTitle?: string;
  onSubmitTitle?: string;
  submitButtonId?: string;
}
/**
 *
 * `PopupForm` is a combination of `Form` and `Popup`.
 *
 */
export const PopupForm = React.forwardRef<HTMLFormElement, PopupFormProps>(
  (props, ref) => {
    const {
      visible,
      onClose,
      onCloseTitle,
      onSubmitTitle,
      className = '',
      submitButtonId,
      children,
      ...rest
    } = props;

    return (
      <Popup visible={visible} className={`tec-popupForm ${className}`}>
        <Form className='tec-popupFormForm' ref={ref} {...rest}>
          <FormContent className='tec-popupFormContent'>{children}</FormContent>
          <ButtonContainer className='tec-popupFormButtonContainer'>
            <LeftButton
              onClick={onClose}
              className='tec-popupFormButton PopupFormLeftButton'
            >
              {onCloseTitle}
            </LeftButton>
            <RightButton
              type='submit'
              className='tec-popupFormButton PopupFormRightButton'
              id={submitButtonId}
            >
              {onSubmitTitle}
            </RightButton>
          </ButtonContainer>
        </Form>
      </Popup>
    );
  },
);

PopupForm.defaultProps = {
  visible: false,
  onCloseTitle: 'Cancel',
  onSubmitTitle: 'Submit',
};
