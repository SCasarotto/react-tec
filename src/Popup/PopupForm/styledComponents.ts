import styled from 'styled-components';

import { Button } from '../../Form/Button';
import { Popup as P } from '../Popup';

export const Popup = styled(P)`
  .rec-popupContent {
    width: 90%;
    max-width: 650px;
    margin: 100px auto;
  }
`;
export const FormContent = styled.div`
  padding: 30px;
`;
export const ButtonContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;
export const LeftButton = styled(Button)`
  border-top: 1px solid ${(props) => props.theme.primary};
  border-right: 1px solid ${(props) => props.theme.primary};
  border-bottom: none;
  border-left: none;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 0px;
  padding-top: 15px;
  padding-bottom: 15px;
  font-size: 18px;

  :active,
  :hover {
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.primary};
    border-top: 1px solid ${(props) => props.theme.primary};
    border-right: 1px solid ${(props) => props.theme.primary};
    border-bottom: none;
    border-left: none;
  }
`;
export const RightButton = styled(Button)`
  border-top: 1px solid ${(props) => props.theme.primary};
  border-right: none;
  border-bottom: none;
  border-left: 1px solid ${(props) => props.theme.primary};
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 20px;
  padding-top: 15px;
  padding-bottom: 15px;
  font-size: 18px;

  :active,
  :hover {
    color: ${(props) => props.theme.white};
    background-color: ${(props) => props.theme.primary};
    border-top: 1px solid ${(props) => props.theme.primary};
    border-right: none;
    border-bottom: none;
    border-left: 1px solid ${(props) => props.theme.primary};
  }
`;
