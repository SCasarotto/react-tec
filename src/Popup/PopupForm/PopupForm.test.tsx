import React from 'react';

import { render, waitFor } from '@testing-library/react';

import { PopupForm } from './PopupForm';

describe('PopupForm', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(<PopupForm visible onClose={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(
      <PopupForm visible onClose={jest.fn()} className='myClassName' />,
    );
    expect(container.firstChild).toHaveClass('myClassName');
  });

  // Ref
  it('renders with ref', async () => {
    const ref: { current: HTMLFormElement | null } = { current: null };
    render(
      <PopupForm
        visible
        onClose={jest.fn()}
        ref={ref}
        data-testid='myFormId'
      />,
    );
    // Using this as the focus check didn't seem to work
    waitFor(() => expect(ref.current).toBeTruthy());
  });
});
