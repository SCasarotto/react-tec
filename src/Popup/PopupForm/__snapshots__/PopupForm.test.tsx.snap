// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupForm matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  position: relative;
  width: 100%;
  margin: 0px auto;
  box-sizing: border-box;
}

.c6 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 18px;
  padding: 5px 20px;
  text-align: center;
  cursor: pointer;
  border-radius: 5px;
  box-sizing: border-box;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: color 0.2s ease-in,background-color 0.2s ease-in, border 0.2s ease-in;
  transition: color 0.2s ease-in,background-color 0.2s ease-in, border 0.2s ease-in;
}

.c6:hover,
.c6:active {
  border: 1px solid;
}

.c6:disabled {
  cursor: not-allowed;
  border: 1px solid;
}

.c0 {
  position: fixed;
  width: 100%;
  height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: rgba(0,0,0,0.5);
  z-index: 1000;
  overflow-y: auto;
  -webkit-transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  transition: visibility 0.25s ease-in-out,opacity 0.25s ease-in-out;
  opacity: 1;
  visibility: visible;
}

.c2 {
  position: relative;
  border-radius: 20px;
  margin: 50px auto;
  max-width: 650px;
}

.c1 .PopupContent {
  width: 90%;
  max-width: 650px;
  margin: 100px auto;
}

.c4 {
  padding: 30px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  border-top: 1px solid;
  border-right: 1px solid;
  border-bottom: none;
  border-left: none;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-left-radius: 20px;
  border-bottom-right-radius: 0px;
  padding-top: 15px;
  padding-bottom: 15px;
  font-size: 18px;
}

.c7:active,
.c7:hover {
  border-top: 1px solid;
  border-right: 1px solid;
  border-bottom: none;
  border-left: none;
}

.c8 {
  border-top: 1px solid;
  border-right: none;
  border-bottom: none;
  border-left: 1px solid;
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 20px;
  padding-top: 15px;
  padding-bottom: 15px;
  font-size: 18px;
}

.c8:active,
.c8:hover {
  border-top: 1px solid;
  border-right: none;
  border-bottom: none;
  border-left: 1px solid;
}

<div
    class="c0 tec-popup c1 tec-popupForm "
  >
    <div
      class="c2 tec-popupContent"
    >
      <form
        class="c3 tec-form tec-popupFormForm"
      >
        <div
          class="c4 tec-popupFormContent"
        />
        <div
          class="c5 tec-popupFormButtonContainer"
        >
          <button
            class="c6 tec-button c7 tec-popupFormButton PopupFormLeftButton"
            type="button"
          >
            Cancel
          </button>
          <button
            class="c6 tec-button c8 tec-popupFormButton PopupFormRightButton"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
